youtube-client-app/src/app/core/services/video-search.service.ts

```diff
@@ -0,0 +1,29 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+import { ISearchItem } from '../../shared/models/search-items.models';
+import youtubeResponse from '../../common/mocks/response';
+import { IResponse } from '../../youtube/models/search-response.models';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class VideoSearchService {
+  public searchVal$ = new Subject<string>();
+
+  public youtubeResponse: IResponse = youtubeResponse;
+
+  public cardCollection!: ISearchItem[];
+
+  public items$ = new Subject<ISearchItem[]>();
```

Comment:
as well as in user case, I think it would be better to have only items$ here, because cardCollection can be derived from items$

Reply:
we can discuss this on our meeting also